src/tests/java/picard/sam/markduplicates/SimpleMarkDuplicatesWithMateCigar.java

```diff
@@ -0,0 +1,211 @@
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * This is a simple tool to mark duplicates using the DuplicateSetIterator, DuplicateSet, and SAMRecordDuplicateComparator.
+ * 
+ * See MarkDuplicates for more details.
+ *
+ * @author nhomer
+ */
+@CommandLineProgramProperties(
+        usage = "Examines aligned records in the supplied SAM or BAM file to locate duplicate molecules. " +
+                "All records are then written to the output file with the duplicate records flagged.",
+        usageShort = "Examines aligned records in the supplied SAM or BAM file to locate duplicate molecules.",
+        programGroup = Testing.class
+)
+public class SimpleMarkDuplicatesWithMateCigar extends MarkDuplicates {
+    private final Log log = Log.getInstance(MarkDuplicatesWithMateCigar.class);
+
+    /** Stock main method. */
+    public static void main(final String[] args) {
+        new MarkDuplicatesWithMateCigar().instanceMainWithExit(args);
+    }
+
+    private class ReadEndsForSimpleMarkDuplicatesWithMateCigar extends ReadEnds {
+    }
+
+    /**
+     * Main work method.
+     */
+    protected int doWork() {
+        IOUtil.assertInputsAreValid(INPUT);
+        IOUtil.assertFileIsWritable(OUTPUT);
+        IOUtil.assertFileIsWritable(METRICS_FILE);
+
+        // Open the inputs
+        final SamHeaderAndIterator headerAndIterator = openInputs();
+        final SAMFileHeader header = headerAndIterator.header;
+
+        // Create the output header
+        final SAMFileHeader outputHeader = header.clone();
+        outputHeader.setSortOrder(SAMFileHeader.SortOrder.coordinate);
+        for (final String comment : COMMENT) outputHeader.addComment(comment);
+        
+        // Key: previous PG ID on a SAM Record (or null).  Value: New PG ID to replace it.
+        final Map<String, String> chainedPgIds = getChainedPgIds(outputHeader);
+
+        // Open the output
+        final SAMFileWriter out = new SAMFileWriterFactory().makeSAMOrBAMWriter(outputHeader,
+                false,
+                OUTPUT);
+        
+        final DuplicateSetIterator iterator = new DuplicateSetIterator(headerAndIterator.iterator,
+                headerAndIterator.header);
+        
+        iterator.setScoringStrategy(this.DUPLICATE_SCORING_STRATEGY);
+
+        // progress logger!
+        final ProgressLogger progress = new ProgressLogger(log, (int) 1e6, "Read");
+
+        int numDuplicates = 0;
+        
+        libraryIdGenerator = new LibraryIdGenerator(headerAndIterator.header);
+        
+        for (final DuplicateSet duplicateSet : new IterableAdapter<DuplicateSet>(iterator)) {
+            final SAMRecord representative = duplicateSet.getRepresentative();
+            final boolean doOpticalDuplicateTracking = representative.getReadPairedFlag() &&
+                    representative.getFirstOfPairFlag() &&
+                    !representative.getReadUnmappedFlag() &&
+                    !representative.getMateUnmappedFlag();
```

Comment:
It might make sense to make a named function for getReadPairedFlag, !getReadUnmappedFlag and !getMateUnmappedFlag, as you use it below as well and it might be nice to label what concept those three together capture.